### PR TITLE
Attempt to fix flaky Spark autologging & model export tests

### PR DESCRIPTION
**Pull request by smurching**
## What changes are proposed in this pull request?

Increases sleep duration for Spark autologging test, & adds retries around initializing a SparkSession for Spark model export tests, as sometimes initializing a SparkSession with the necessary dependent libraries (mleap) can fail due to timeouts while fetching mleap JARs.

Also makes cosmetic fixes to reduce build log length

## How is this patch tested?

Existing tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/spark/test_spark_model_export.py [hunk 1]
```diff
@@ -59,12 +61,19 @@ def spark_context():
              value='ml.combust.mleap:mleap-spark-base_2.11:0.12.0,'
                    'ml.combust.mleap:mleap-spark_2.11:0.12.0')
     conf.set(key="spark_session.python.worker.reuse", value=True)
-    spark = pyspark.sql.SparkSession.builder\
-        .config(conf=conf)\
-        .master("local-cluster[2, 1, 1024]")\
-        .getOrCreate()
-    sc = spark.sparkContext
-    return sc
+    max_tries = 3
```
Comment (smurching) [inline tests/spark/test_spark_model_export.py]: Seems like SparkContext initialization can flake while attempting to fetch mleap libraries (specified as part of the Spark conf [here](https://github.com/mlflow/mlflow/pull/2373/files#diff-9b8029c7d29cba0c31d59382d11e876dR60)), so added retries

File: tests/spark_autologging/test_spark_datasource_autologging_crossframework.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def _fit_keras(pandas_df, epochs):
     keras_model.add(Dense(1))
     keras_model.compile(loss='mean_squared_error', optimizer='SGD')
     keras_model.fit(x, y, epochs=epochs)
-    time.sleep(2)
+    time.sleep(5)
```
Comment (smurching) [inline tests/spark_autologging/test_spark_datasource_autologging_crossframework.py]: Increasing this sleep duration to reduce the odds of runs that fit Keras models in these tests ending before we have a chance to log Spark datasource info. I can't seem to replicate this failure locally but I guess Spark on Travis is slow...

File: travis/test-spark-autologging.sh [hunk 3]
```diff
@@ -6,7 +6,7 @@ set -ex
 
 # Build Java package
 pushd mlflow/java/spark
-mvn package -DskipTests
+mvn package -DskipTests -q
```
Comment (smurching) [inline travis/test-spark-autologging.sh]: Build package silently (only print errors) to reduce build log size

File: travis/test-spark-autologging.sh [hunk 4]
```diff
@@ -16,7 +16,7 @@ popd
 TEMPDIR=$(mktemp -d)
 pushd $TEMPDIR
 wget https://archive.apache.org/dist/spark/spark-3.0.0-preview/spark-3.0.0-preview-bin-hadoop2.7.tgz -O /tmp/spark.tgz
-tar -xvf /tmp/spark.tgz
+tar -xf /tmp/spark.tgz
```
Comment (smurching) [inline travis/test-spark-autologging.sh]: Don't use verbose flag when extracting the tgz (it needlessly prints out all the extracted files & increases build log length)

File: tests/spark_autologging/test_spark_datasource_autologging_crossframework.py (hunk 2 above)
Comment (dbczumar) [reply tests/spark_autologging/test_spark_datasource_autologging_crossframework.py]: Can we add an explanatory comment here indicating why we're sleeping and that this is a known edge case (as documented here: https://mlflow.org/docs/latest/python_api/mlflow.spark.html#mlflow.spark.autolog).

File: travis/test-spark-autologging.sh (hunk 3 above)
Comment (dbczumar) [reply travis/test-spark-autologging.sh]: Awesome!

File: travis/test-spark-autologging.sh (hunk 4 above)
Comment (dbczumar) [reply travis/test-spark-autologging.sh]: Thank you!

**Review by dbczumar — APPROVED** [review]
LGTM once comment about `time.sleep()` is added. Thanks @smurching
